src/backend/feed/worker.js

```diff
@@ -25,7 +27,39 @@ function getFeedWorkersCount() {
   return Math.min(count, cpuCount);
 }
 
-exports.start = function() {
+exports.start = async function() {
+  /**
+   * Elasticsearch needs time after deployment for setting up all its components.
```

Comment:
I'm curious about this.  The elastic docs don't show this as being necessary.  Obviously you've read something I haven't, but it surprises me that we need to do this synchronous dance around a new connection.  Can you point me at docs to understand why this is necessary?  I wonder if the older JS client needed this and the new Promise-based on doesn't?

Reply:
When I first started testing I noticed how I was trying to index posts and failing because elasticsearch wasn't ready. The idea of waiting for elasticsearch to be ready is from [here](https://blog.logrocket.com/full-text-search-with-node-js-and-elasticsearch-on-docker/) (search `function checkconnection()` in the page).
> Now let’s consider that ElasticSearch is a huge system and can take some seconds to boot up. We can’t connect to ES until it is ready, so we need a function that checks when the ES server is ready:

```js
function checkConnection() {
  return new Promise(async (resolve) => {
    console.log("Checking connection to ElasticSearch...");
    let isConnected = false;
    while (!isConnected) {
      try {
        await esclient.cluster.health({});
        console.log("Successfully connected to ElasticSearch");
        isConnected = true;
      // eslint-disable-next-line no-empty
      } catch (_) {
      }
    }
    resolve(true);
  });
}
```
They use an infinite loop to wait for ES to be ready, but I thought it was better to have a timeout in case ES failed (without a timeout, if ES is down telescope will never launch).

Reply:
I read [a bit more into this](https://docs.docker.com/compose/startup-order/) and probably you're right that we make our dependent services something we wait on before we assume they are running and can be used.  I'll file a follow-up to do the same for Redis, since we should be consistent.

I wonder if we should actually crash if elastic isn't there (same for Redis).  Likely search will become more than a nice-to-have once we have it.

Reply:
Can you move this whole block of code (down to line 61) into your elastic client code, and call it `waitOnReady()` or `init()` or something, and have it return a `Promise`?  That would simplify this code to:

```js
exports.start = async function() {
  try {
    await Promise.all([elastic.waitOnReady(), redis.waitOnReady()])
    // keep going with starting the workers here...
  } catch(err) {
    // Redis, Elastic, or both failed, we need to crash...
  }
}
```